### PR TITLE
Include new warning codes in --notrimwarn and --noaotwarn

### LINKED ISSUE #105044: Trimmer reports IL2122 even if trim warnings are suppressed
```csharp
using System.Diagnostics.CodeAnalysis;

RequireAll("NonExistent");

void RequireAll([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] string typeName)
{
    Type.GetType(typeName);
}
```

```
> dotnet publish /p:PublishTrimmed=true /p:SuppressTrimAnalysisWarnings=true
ILLink: warning IL2122: Program.<Main>$(String[]): Type 'NonExistent' is not assembly qualified. Type name strings used for dynamically accessing a type should be assembly qualified.
```

This is because:
https://github.com/dotnet/runtime/blob/57f870f909dbfad35142e5aaa6e681464de4f439/src/tools/illink/src/ILLink.Shared/DiagnosticId.cs#L230

Thew newly added warning is not part of any of the recognized ranges.

### PR DESCRIPTION
**Pull request by sbomer**
Fixes https://github.com/dotnet/runtime/issues/105044

### CODE REVIEW

File: src/tools/illink/test/ILLink.RoslynAnalyzer.Tests/WarningsTests.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+using System;
```
Comment (MichalStrehovsky) [inline src/tools/illink/test/ILLink.RoslynAnalyzer.Tests/WarningsTests.cs:1]: Copyright header?

File: src/tools/illink/test/Mono.Linker.Tests.Cases/Warnings/CanDisableWarningsByCategory.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
+using System.Diagnostics.CodeAnalysis;
```
Comment (MichalStrehovsky) [inline src/tools/illink/test/Mono.Linker.Tests.Cases/Warnings/CanDisableWarningsByCategory.cs:1]: Header?

**Review by MichalStrehovsky — APPROVED** [review]

File: src/tools/illink/src/ILLink.Shared/DiagnosticId.cs [hunk 3]
```diff
@@ -227,9 +229,9 @@ public static class DiagnosticIdExtensions
 				2103 => MessageSubCategory.TrimAnalysis,
 				2106 => MessageSubCategory.TrimAnalysis,
 				2107 => MessageSubCategory.TrimAnalysis,
-				>= 2109 and <= 2121 => MessageSubCategory.TrimAnalysis,
+				>= 2109 and <= (int) DiagnosticId._EndTrimAnalysisWarningsSentinel => MessageSubCategory.TrimAnalysis,
```
Comment (vitek-karas) [inline src/tools/illink/src/ILLink.Shared/DiagnosticId.cs:232]: Technically this should be just `<`

File: src/tools/illink/src/ILLink.Shared/DiagnosticId.cs [hunk 4]
```diff
@@ -187,6 +187,7 @@ public enum DiagnosticId
 		_unused_DynamicallyAccessedMembersOnTypeReferencesCompilerGeneratedMemberOnBase = 2120,
 		RedundantSuppression = 2121,
 		TypeNameIsNotAssemblyQualified = 2122,
+		_EndTrimAnalysisWarningsSentinel = 2123,
```
Comment (vitek-karas) [inline src/tools/illink/src/ILLink.Shared/DiagnosticId.cs:190]: Would it be better to not specify the value - AFAIK C# should assign the next (+1) value automatically which would make this work nicely when we add a new one.

**Pull request merged**